device/core/test/_internal_client_test.js

```diff
@@ -172,31 +172,35 @@ var ModuleClient = require('../lib/module_client').ModuleClient;
         fs.unlinkSync('aziotfakepemfile');
       });
 
-      /*Tests_SRS_NODE_INTERNAL_CLIENT_06_001: [The `setOption` method shall first test if the `ca` property is the name of an already existent file.  If so, it will attempt to read that file as a pem into a string value and pass the string to config object `ca` property.  Otherwise, it is assumed to be a pem string.] */
+      /*Tests_SRS_NODE_INTERNAL_CLIENT_06_001: [The `setOptions` method shall assume the `ca` property is the name of an already existent file and it will attempt to read that file as a pem into a string value and pass the string to config object `ca` property.  Otherwise, it is assumed to be a pem string.] */
       it('sets CA cert with contents of file if provided', function (testCallback) {
         var fakeBaseClient = new FakeTransport();
-        fakeBaseClient.setOptions = sinon.stub();
+        fakeBaseClient.setOptions = sinon.stub().callsArg(1);
         var fakeMethodClient = {}
         fakeMethodClient.setOptions = sinon.stub();
         var client = new ClientCtor(fakeBaseClient);
         client._methodClient = fakeMethodClient;
-        client.setOptions({ ca: 'aziotfakepemfile' });
-        assert(fakeBaseClient.setOptions.called);
-        assert.strictEqual(fakeBaseClient.setOptions.firstCall.args[0].ca, 'ca cert');
-        testCallback();
+        client.setOptions({ ca: 'aziotfakepemfile' }, function(err) {
+          assert.isNotOk(err, 'the setOptions passed')
```

Comment:
>'the setOptions passed' [](start = 30, length = 23)

nit: doesn't mean anything to me. this assumes we can see the code of the test, which is not how test reports are used. this is going to be printed if the test fails, in a test report that doesn't show the code. we almost never use these as i don't feel it helps the test report and it's counterintuitive to read in the code but if you want to start democratizing this practice at least let's make sure the error describes explicitely why the assertion failed.

Reply:
removed.

---
In reply to: [221353834](https://github.com/Azure/azure-iot-sdk-node/pull/378#discussion_r221353834) [](ancestors = 221353834)